tests/e2e_tests/test_post_algorithms.py

```diff
@@ -7,117 +7,135 @@
 
```

Comment:
Why are there changes in this file? What changed on the api?

Reply:
I think that for e2e tests we need to have hardcoded json strings. The reason is that if we change something there, in order for tests to pass, it means that the integration with the portal backend will break, so we need to know that. If we cast it into a pydantic object and then jsonify that object, maybe when changing that object, the test will pass but then the integration will fail and we won't notice it.

Reply:
Maybe not hardcoded json strings if there is a better way but you get the point...